mono/metadata/object-internals.h

```diff
@@ -263,6 +263,7 @@ struct _MonoReflectionType {
 	MonoType  *type;
 };
 
+/* This corresponds to System.RuntimeType */
 typedef struct {
 	MonoReflectionType type;
 	MonoObject *type_info;
```

Comment:
I wonder if we can remove `MonoReflectionMonoType`.
At the time we started using `System.RuntimeType`, we broke any API client that tried to use this `type_info` field (because `System.RuntimeType:ObjectCache` was laid out at that offset).  Assuming we didn't get any bug reports about it, that seems like pretty good evidence that this was never used.